### PR TITLE
Resolves #25 - handle non-nested addon options

### LINKED ISSUE #25: not picking up custom 'babel' options in 0.2.0 ember-cli
I am trying to prevent the default `compact` behavior of babel

I set 
```
new EmberApp({
    babel: {
        compact: false
    },
```
that option is never seen in the invocation of babel

I'm logging this.parent.options in setupPreprocessorRegistry and it's undefined.

logging this while I look for correct solution

### PR DESCRIPTION
**Pull request by jschilli**
This works but feels a tad dirty.

`setupPreprocessorRegistry` is called super early in the process so I capture the app in included - which does make app.options available before the tree is processed.

I'm open to refinements and suggestions here if there is a cleaner path.

Seems like an addon method to determine if its nested or not would be a nice addition

cc: @lukemelia @stefanpenner

### CODE REVIEW

**Comment by rwjblue** [comment]
`this.parent` is intended to be the project for non-nested addons (aka direct project dependencies), and the parent addon in the nested addon case.  Unfortunately, the current config story kinda sucks right now and there is no way to provide build time configuration to the `project` instance without polluting the runtime config (in `index.html`'s `<meta>`).

This fix is good for now, but we need to fix the config story to get a path forward to avoiding the fallback lookup added here.

**tldr;** :+1:

File: index.js [hunk 1]
```diff
@@ -8,13 +8,12 @@ module.exports = {
   },
 
   setupPreprocessorRegistry: function(type, registry) {
-    var options = getOptions(this.parent && this.parent.options && this.parent.options['babel']);
-
+    var that = this;
```
Comment (stefanpenner) [inline index.js]: lets call this `addon` to be clear what it is.

**Comment by jschilli** [comment]
not sure what happened with that build

**Comment by stefanpenner** [comment]
rebuilding....

**Comment by jschilli** [comment]
I'm going to push a new branch to see if this will build cleanly - not sure what happened - my force push to my branch (to squash) may have bungled it
